pkg/util/winutil/users.go

```diff
@@ -75,3 +76,29 @@ func GetLocalSystemSID() (*windows.SID, error) {
 
 	return localSystem, err
 }
+
+// GetDataDogAgentSID returns the SID of the DataDog Agent account
+func GetDDAgentUserSID() (*windows.SID, error) {
+	// get config for datadogagent service
+	user, err := GetServiceUser("datadogagent")
+	if err != nil {
+		return nil, fmt.Errorf("could not get datadogagent service user: %s", err)
+	}
+
+	var domain, username string
+	parts := strings.SplitN(user, "\\", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("could not parse user: %s", user)
+	}
+
+	domain = parts[0]
+	if domain == "." {
+		username = parts[1]
+	} else {
+		username = user
+	}
+
+	// get the SID for the user account
+	sid, _, _, err := windows.LookupSID("", username)
```

Comment:
Can you break this up into a function like `GetServiceUserSID(service string)` so that you can add some tests to see if you can get the SID for some built-in/default services that run as `LocalSystem` and `LocalService` to ensure those resolve correctly? In case the Agent is running as those users. SCM has weird aliases that only it recognizes and I'm not sure what the config API will return
https://learn.microsoft.com/en-us/windows/win32/services/service-user-accounts

Reply:
added